app/medInria/medMainWindow.cpp

```diff
@@ -674,18 +674,40 @@ void medMainWindow::hideShortcutAccess()
     this->activateWindow();
 }
 
-int medMainWindow::saveModified( void )
+int medMainWindow::saveModifiedAndOrValidateClosing()
 {
     QList<medDataIndex> indexes = medDatabaseNonPersistentController::instance()->availableItems();
 
     if(indexes.isEmpty())
-        return QDialog::Accepted;
+    {
+        // No data to save, pop-up window to validate the closing
+
+        QMessageBox msgBox;
+        msgBox.setWindowTitle("Closing");
+        msgBox.setText("Do you really want to exit?");
+        msgBox.setStandardButtons(QMessageBox::Yes);
+        msgBox.addButton(QMessageBox::No);
+        msgBox.setDefaultButton(QMessageBox::No);
+        if(msgBox.exec() == QMessageBox::Yes)
+        {
+            return QDialog::Accepted;
+        }
+        else
+        {
+            return QDialog::Rejected;
+        }
+    }
+    else
+    {
+        // User is asked to save, cancel or exit without saving temporary data
 
-    medSaveModifiedDialog *saveDialog = new medSaveModifiedDialog(this);
-    saveDialog->show();
-    saveDialog->exec();
+        medSaveModifiedDialog *saveDialog = new medSaveModifiedDialog(this);
+        saveDialog->show();
+        saveDialog->exec();
+        return saveDialog->result();
+    }
 
-    return saveDialog->result();
+    return QDialog::Rejected;
```

Comment:
This line cannot be reached and may generate warnings on some compilers. It should probably be removed.

Reply:
in fact this line is added to avoid warnings on compilers. We had to do it in some of our functions too in musicPlugins

Reply:
There were cases where we had errors related to functions that didn't return on all possible paths, but here it's the opposite, it is code that is unreachable. There are many functions in MUSIC that have functions like the example below without the compiler complaining:
```
int foo ()
{
   if (blah)
   {
      return 1;
   }
   else
   {
      return 0;
   }
// no return here
}
```